app/controllers/dashboards_controller.rb

```diff
@@ -97,12 +98,25 @@ def subscriptions
       .includes(:subscriber).order(created_at: :desc).page(params[:page]).per(100)
   end
 
+  def hidden_tags
+    fetch_and_authorize_user
+    @hidden_tags = follows_for_tag(user: @user, order_by: :points, points_query: "explicit_points < 0")
+    @collections_count = collections_count(@user)
+  end
+
   private
 
   def follows_for(user:, type:, order_by: :created_at)
     user.follows_by_type(type).order(order_by => :desc).includes(:followable).limit(follows_limit)
   end
 
+  def follows_for_tag(user:, order_by: :created_at, points_query: "explicit_points >= 0")
+    user.follows_by_type("ActsAsTaggableOn::Tag").order(order_by => :desc)
```

Comment:
I think we could possibly use the Tag resource to render the page using `Tag.followed_by(user).order(order_by => :desc).limit(follows_limit)`. Once we get rid of the weights/points on the page, we might able to use this query as we will not need the `points` from the `Follow` resource. We could also improve on the followings_controller query and maybe abstract out some of code. I didn't do that as yet because we we're still shaping the feature and once its in a more refined/final state I can work on the improving re-use.

Reply:
Yeah we are removing user-assigned points from tags entirely based on the [figma](https://www.figma.com/file/nj32OuyDOR430frESjcOCM/Tag-Interaction?type=design&node-id=1637-1693&mode=design&t=dnovOzpWzISLBVPn-0) here.